### ISSUE TITLE
chalice deploy import error

### ISSUE THREAD

**Issue by vfreeclc** [coinbase/coinbase-advanced-py#76]
I have a chalice app that runs locally and returns my coinbase accounts data.  However chalice deploy loads but when requesting accounts info I get :

Runtime.ImportModuleError: Unable to import module 'app': No module named 'coinbase.rest'

I have tried many simple chalice apps and it seems that local can find modules in subdirectories but deploy can't.
So far I have not been able to resolve this through chalice documentation or internet searching or AI. 
I can't get past the coinbase.rest issue to see if cryptography will work deployed, which has many subdirectories.
I have coinbase in my requirements.txt.  I've no idea how much detail to post here.  Hoping someone else has come across this before.

**Comment by kanugaba-cb** [comment]
This often indicates a difference in how dependencies are managed between your local environment and the deployed Lambda function. Here are a few things you can try out:

1. Upgrade to the latest version of the Coinbase Advanced API Python SDK: `pip3 install --upgrade coinbase-advanced-py`
2. Update requirements.txt to specify the correct package name and version: `coinbase-advanced-py==1.7.0`
3. Check that your import statement is correct: `from coinbase.rest import RESTClient`
4. Check that your Chalice configuration file, `.chalice/config.json` is correctly set up for packaging dependencies.
5. Try to explicitly vendor your dependencies. Create a `vendor` directory in your project root and install the packages there: `pip3 install -t ./vendor coinbase-advanced-py`. Then add to your `.chalice/config.json`:
`{
  "app_name": "your-app-name",
  "stages": {
    "dev": {
      "api_gateway_stage": "api",
      "environment_variables": {
        "PYTHONPATH": "./vendor"
      }
    }
  }
}`
6. After deploying check that your Lambda function configuration in the AWS console has enough memory and timeout to accommodate loading these libraries.

**Comment by vfreeclc** [comment]
Thank you for responding. A clean install of everything in a new venv got chalice to access coinbase-advanced-py.  It was confusing as jwt and cryptography seemed to work with an empty requirements file. More confusion came when chalice was unable to find files I created and tried to import. An hour after posting the first time here I came across a video that mentioned the chalicelib directory. Solved all my issues.  I know this was an issue more appropriately posted to a chalice site. But someone may find this post helpful within the coinbase community.
